core/src/main/scala/kafka/log/Log.scala

```diff
@@ -434,13 +499,23 @@ class Log(@volatile var dir: File,
           maxOffsetInMessages = appendInfo.lastOffset)
 
 
-        // now append to the log
+        // update the first offset if on the first message. For magic versions older than 2, we use the last offset
```

Comment:
Not sure how that moved around like that.  I think it must have been a fat fingered cut/paste from last night. Moved it back now.